src/Common/src/System/Text/ConsoleEncoding.cs

```diff
@@ -4,6 +4,19 @@
 
 namespace System.Text
 {
+
+    internal static class EncodingExtensions
+    {
+        public static Encoding RemovePreamble(this Encoding encoding)
+        {
+            if (encoding.GetPreamble().Length == 0)
```

Comment:
Can the new `ReadOnlySpan<byte>` returning `Preamble` property be used here instead (to avoid the unnecessary array allocation)?